pkgs/development/libraries/mbedtls/generic.nix

```diff
@@ -47,6 +47,10 @@ stdenv.mkDerivation rec {
     "-DGEN_FILES=off"
   ];
 
+  env = lib.optionalAttrs stdenv.cc.isGNU {
+    NIX_CFLAGS_COMPILE = "-Wno-error=calloc-transposed-args";
```

Comment:
This is a warning turned into an error back to not being an error; it would have been as dire on gcc 13 as with gcc 14.

/cc mbedtls nixpkgs maintainers: @raphaelr

Reply:
AFAIK this should only affect `mbedtls_2`, not `mbedtls` (v3). This issue has already been fixed for mbedtls v3: https://github.com/Mbed-TLS/mbedtls/pull/8751, but the fix wasn't backported to v2.

IMO the cleanest solution would be to apply that patch manually for v2, and leave v3 unchanged. It's unlikely that this patch will cause conflicts for v2 in the future since that branch only receives bugfixes/maintenance updates.
```nix
# pkgs/development/libraries/mbedtls/2.nix
{ callPackage
, fetchpatch
}:

callPackage ./generic.nix {
  version = "2.28.9";
  hash = "sha256-/Bm05CvS9t7WSh4qoMconCaD7frlmA/H9YDyJOuGuFE=";

  patches = [
    (fetchpatch {
      name = "gcc14-fix-calloc-argument-list.patch";
      url = "https://github.com/Mbed-TLS/mbedtls/commit/da2a33de0f2d0f7f179220d1a92f788d1459c3e3.patch";
      hash = "sha256-iKi1G0UDM2TlmVnSR8lRAkcQHnz7SQzJNQ3z2hAwZnw=";
    })
  ];
}
```

Reply:
It's just a style nit, not sure it's worth patching over.

Reply:
I have no objections to the initial approach.

Reply:
Thanks for the discussion. I'll go with the team.